apps/src/lib/kits/maker/BoardController.js

```diff
@@ -134,10 +134,9 @@ export default class BoardController {
     if (!this.board_) {
       return;
     }
-    // TouchSensors aren't in board_.register because they are our own wrapper
-    // around TouchPad, but we still need to reset them.
-    const touchSensors = _.filter(this.prewiredComponents, c => c instanceof TouchSensor);
-    this.board_.register.concat(touchSensors).forEach(BoardController.resetComponent);
+    this.board_.io.reset();
+    this.board_ = null;
+    this.prewiredComponents = null;
```

Comment:
Good point, maybe I'll rename this to `destroy` during followup refactoring.

`BoardController`'s [`connectAndInitialize()`](https://github.com/code-dot-org/code-dot-org/blob/765c70a23a6de15d32064c6553111210d7c486bd/apps/src/lib/kits/maker/BoardController.js#L41-L46) is already set up to handle both the first run and subsequent run cases, by following a lazy-create pattern; now that we are destroying the board and components here, it will always follow it's first-run path and reinitialize them every time.